src/lib-sana/Graph/Graph.cpp

```diff
@@ -34,10 +34,16 @@ int Graph::RandomNode() {
 void Graph::AddEdge(const unsigned int &node1, const unsigned int &node2, const unsigned int &weight) throw(GraphInvalidIndexError) {
     if (node1 >= numNodes || node2 >= numNodes)
         throw GraphInvalidIndexError("Invalid node index passed into AddEdge");
-
-    adjLists[node1].push_back(node2);
-    adjLists[node2].push_back(node1);
-    ++numEdges;
+    if (node1 > node2) {
+        unsigned int node1_cpy = node2;  // Const stuff

```

Comment:
can we just do `adjLists[node2].push_back(node1);` ?

Reply:
Absolutely, go ahead and change it. I'm not at home today.

I don't know what I was thinking.